### PR TITLE
[DDL] Enable DDL to show different claim letters from other tools

### PR DESCRIPTION
**Pull request by samcoforma**
## Summary

- Modified `ClaimLetterDownloader` to accept `allowed_doctypes` as a parameter on its initializer
- Updated `claim_letter_downloader_spec.rb`
- Modified `ClaimLettersController` to pass `allowed_doctypes` to `ClaimLetterDownloader`
- Modified mobile team's `DecisionLettersController` to pass `allowed_doctypes` to `ClaimLetterDownloader`
- Created new `VirtualAgentClaimLettersController` for Virtual Agent team
- Created `spec/controllers/v0/virtual_agent_claim_letters_controller_spec.rb` RSpec test for new `VirtualAgentClaimLettersController`
- Created new routes:
   - `virtual_agent_claim_letters`
   - `virtual_agent_claim_letters/:document_id`

## Related issue(s)

- [va.gov-team/issues/72958](https://github.com/department-of-veterans-affairs/va.gov-team/issues/72958)

## Testing done

- [x] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by aherzberg — DISMISSED** [review]
LGTM for mobile changes. Thanks!

File: lib/claim_letters/claim_letter_downloader.rb [hunk 1]
```diff
@@ -66,12 +61,17 @@ def file_number
     end
 
     def filter_letters(document)
-      return nil unless allowed_doctypes.include?(document[:doc_type])
+      return nil unless @allowed_doctypes.include?(document[:doc_type])
 
-      document.marshal_dump
+      if document.is_a?(OpenStruct)
```
Comment (jerekshoe) [inline lib/claim_letters/claim_letter_downloader.rb:66]: What caused this change? Did you run into an error with the previous implementation?

Comment (samcoforma) [reply lib/claim_letters/claim_letter_downloader.rb:66]: I added a check for `filter_letters(letter_details).nil?` in the `get_letter` method, and needed `filter_letters` to start accepting non-`OpenStruct` objects for it to work. @jerekshoe

Comment (samcoforma) [reply lib/claim_letters/claim_letter_downloader.rb:66]: Adding that check resolves the open question in the ticket for this: `Should we be worried about filtering out based on docTypes when requesting documents by their ID?`

Comment (jerekshoe) [reply lib/claim_letters/claim_letter_downloader.rb:66]: This makes me wonder if we should move the marshal_dump stuff out of this method. I would at least add a comment about why we need to check

Comment (samcoforma) [reply lib/claim_letters/claim_letter_downloader.rb:66]: @jerekshoe Ok, I'll refactor and/or comment on it. Thanks.

**Review by ericboehs — DISMISSED** [review]

File: app/controllers/v0/claim_letters_controller.rb [hunk 2]
```diff
@@ -24,7 +24,18 @@ def show
...
+
+    # 27: Board Of Appeals Decision Letter
+    # 184: Notification Letter (e.g. VA 20-8993, VA 21-0290, PCGL)
+    # 339: Rating Decision Letter
+    def allowed_doctypes
+      if Flipper.enabled?(:cst_include_ddl_boa_letters, @user)
+        %w[27 184]
+      else
+        %w[184]
+      end
```
Comment (ksk385) [inline app/controllers/v0/claim_letters_controller.rb:38]: So on the existing route this PR will leave the doc types the same, right?

Comment (ksk385) [reply app/controllers/v0/claim_letters_controller.rb:38]: After this PR you will have another PR for updating doc types internally. And before that PR, we need update our bot skill to point to the new route from below. right?

Comment (samcoforma) [reply app/controllers/v0/claim_letters_controller.rb]: @ksk385 The existing route and controller will still work the way it was before, so we don't need any additional PRs. There is a new controller (`app/controllers/v0/virtual_agent_claim_letters_controller.rb`) and route (`config/routes.rb:158`) in this PR for your use when you want to switch your skill over to start using it. I don't see any need to have multiple PRs given how this is structured. Thoughts?

Comment (samcoforma) [reply app/controllers/v0/claim_letters_controller.rb]: @ksk385 I understand your concern here, but to clarify: the changes to the existing controller are merely a refactor and decoupling of where the control over allowed_doctypes exists. It doesn't actually change any of the existing functionality, but sets up the ability to independently control allowed_doctypes from different controllers. It won't affect how the existing route works or what it returns in the short term. Does that make sense? cc: @jerekshoe

Comment (ksk385) [reply app/controllers/v0/claim_letters_controller.rb]: Yes makes sense! Thank you!

**Review by jerekshoe — APPROVED** [review]

**Review by aherzberg — APPROVED** [review]
LGTM

**Review by ericboehs — APPROVED** [review]

**Pull request merged**